build-system/common/esbuild-babel.js

```diff
@@ -77,42 +141,45 @@ function getEsbuildBabelPlugin(
     return read;
   }
 
-  function transformContents(filepath, contents, hash) {
+  async function transformContents(contents, hash, babelOptions) {
     if (enableCache) {
-      const cached = transformCache.get(filepath);
-      if (cached && cached.hash === hash) {
```

Comment:
Nit: I actually keyed it on `filepath` so we wouldn't have unbounded growth of the cache. We'd only store 1 copy of the file. Not huge, but something to think about.

Reply:
Its a tradeoff! By _not_ having filepath, we can reuse cache between multiple branches

Reply:
Some of the code in https://github.com/ampproject/amphtml/blob/cc13d05690d477369e364bc8d6275d126d190a1b/build-system/common/browserify-cache.js might be reusable here. (Also a follow up.)